lib/plugins.js

```diff
@@ -533,8 +552,9 @@ function plugins (settings, deviceId) {
       _.mapKeys(_.snakeCase)
     )
     const tradeEntry = massage(_tradeEntry, error)
-    const sql = pgp.helpers.insert(tradeEntry, null, 'trades')
-    return db.none(sql)
+    const sql = pgp.helpers.insert(tradeEntry, null, 'trades') + 'RETURNING *'
+    return db.oneOrNone(sql)
+      .then(newTrade => recordTradeAndTx(newTrade.id, _tradeEntry.internalTxIdList))
```

Comment:
This also looks like chained queries, if they are, do them inside a task or a tx (transaction) https://vitaly-t.github.io/pg-promise/Task.html